x/ccv/provider/keeper/power_shaping.go

```diff
@@ -551,3 +550,121 @@ func (k Keeper) DeleteMinimumPowerInTopN(
+}
+
+// GetPriorityList returns all prioritylisted validators
+func (k Keeper) GetPriorityList(
+	ctx sdk.Context,
+	consumerId string,
+) (providerConsAddresses []types.ProviderConsAddress) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.StringIdWithLenKey(types.PrioritylistKeyPrefix(), consumerId)
+	iterator := storetypes.KVStorePrefixIterator(store, key)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		providerConsAddresses = append(providerConsAddresses, types.NewProviderConsAddress(iterator.Key()[len(key):]))
+	}
+
+	return providerConsAddresses
+}
+
+// IsPrioritylisted returns `true` if validator with `providerAddr` has been prioritylisted on chain `consumerId`
+func (k Keeper) IsPrioritylisted(
+	ctx sdk.Context,
+	consumerId string,
+	providerAddr types.ProviderConsAddress,
+) bool {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.PrioritylistKey(consumerId, providerAddr))
+	return bz != nil
+}
+
+// DeletePrioritylist deletes all prioritylisted validators
+func (k Keeper) DeletePrioritylist(ctx sdk.Context, consumerId string) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := storetypes.KVStorePrefixIterator(store, types.StringIdWithLenKey(types.PrioritylistKeyPrefix(), consumerId))
+	defer iterator.Close()
+
+	keysToDel := [][]byte{}
+	for ; iterator.Valid(); iterator.Next() {
+		keysToDel = append(keysToDel, iterator.Key())
+	}
+
+	for _, key := range keysToDel {
+		store.Delete(key)
+	}
+}
+
+// IsPrioritylistEmpty returns `true` if no validator is prioritylisted on chain `consumerId`
+func (k Keeper) IsPrioritylistEmpty(ctx sdk.Context, consumerId string) bool {
+	store := ctx.KVStore(k.storeKey)
+	iterator := storetypes.KVStorePrefixIterator(store, types.StringIdWithLenKey(types.PrioritylistKeyPrefix(), consumerId))
+	defer iterator.Close()
+
+	return !iterator.Valid()
+}
+
+// UpdatePrioritylist populates the prioritylist store for the consumer chain with this consumer id
+func (k Keeper) UpdatePrioritylist(ctx sdk.Context, consumerId string, prioritylist []string) {
+	k.DeletePrioritylist(ctx, consumerId)
+	for _, address := range prioritylist {
+		consAddr, err := sdk.ConsAddressFromBech32(address)
+		if err != nil {
+			continue
+		}
+
+		k.SetPrioritylist(ctx, consumerId, types.NewProviderConsAddress(consAddr))
+	}
+}
+
+// FilterAndSortPriorityList filters the priority list to include only validators that can validate the chain
+// and splits the validators into priority and non-priority sets.
+func (k Keeper) FilterAndSortPriorityList(ctx sdk.Context, consumerId string, nextValidators []types.ConsensusValidator) ([]types.ConsensusValidator, []types.ConsensusValidator) {
+	validatorMap := make(map[string]types.ConsensusValidator)
+	for _, v := range nextValidators {
+		validatorMap[string(v.ProviderConsAddr)] = v
+	}
+
+	priorityValidators := make([]types.ConsensusValidator, 0)
+	nonPriorityValidators := make([]types.ConsensusValidator, 0)
+	addedAddresses := make(map[string]bool)
```

Comment:
`addedAddresses` seems too generic, maybe `isPrioritylisted`, `isPrioritylistedValidator`, or `prioritylistedValidators` would better convey how this `map` is used.

Reply:
I removed it now since it's not needed.